### PR TITLE
feat: add cli generate command

### PR DESCRIPTION
**Pull request by owlas**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #2143 

### Scope of the changes (select all that apply):
- [ ] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

File: packages/cli/src/dbt/context.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    try {
+        config = yaml.load(
+            readFileSync(projectFilename, { encoding: 'utf-8' }),
+        ) as any;
+    } catch (e) {
+        throw new ParseError(
+            `Is ${projectDir} a valid dbt project? Couldn't find a valid dbt_project.yml file:\n${e.message}`,
+        );
+    }
+    const targetSubDir = config['target-path'] || './target';
```
Comment (ZeRego) [inline packages/cli/src/dbt/context.ts:30]: Should we support target-dir and models-dir ? We support it in `dbtCliClient.ts` 

`config['target-path'] || config['target-dir'] || '/target',`

File: packages/cli/src/dbt/context.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}: GetDbtContextArgs): DbtContext => {
+    const projectFilename = path.join(projectDir, 'dbt_project.yml');
+    let config;
+    try {
+        config = yaml.load(
+            readFileSync(projectFilename, { encoding: 'utf-8' }),
+        ) as any;
+    } catch (e) {
+        throw new ParseError(
+            `Is ${projectDir} a valid dbt project? Couldn't find a valid dbt_project.yml file:\n${e.message}`,
```
Comment (ZeRego) [inline packages/cli/src/dbt/context.ts:27]:
```suggestion
            `Is ${projectDir} a valid dbt project? Couldn't find a valid dbt_project.yml file at ${projectFilename}:\n${e.message}`,
```

File: packages/cli/src/dbt/profile.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            `Could not find a valid profiles.yml file at ${profilePath}:\n  ${e.message}`,
+        );
+    }
+    const profile = allProfiles[profileName];
+    if (!profile) {
+        throw new ParseError(
+            `Profile ${profileName} not found in ${profilePath}`,
+        );
+    }
+    const target = profile.outputs[targetName || profile.target];
```
Comment (ZeRego) [inline packages/cli/src/dbt/profile.ts:32]: Should we throw an error if `target` is undefined ? e.g they typed the target name wrong ?

File: packages/cli/src/dbt/targets/bigquery.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
+import { JSONSchemaType } from 'ajv';
```
Comment (ZeRego) [inline packages/cli/src/dbt/targets/bigquery.ts:1]: Should we remove this from this PR or do we want to support bigquery in this PR ?

Comment (owlas) [reply packages/cli/src/dbt/targets/bigquery.ts]: deleted!

File: packages/cli/src/dbt/context.ts (hunk 1 above)
Comment (owlas) [reply packages/cli/src/dbt/context.ts:30]: This was recently renamed! I'll update this

Comment (owlas) [reply packages/cli/src/dbt/context.ts:30]: https://docs.getdbt.com/reference/dbt_project.yml

**Review by ZeRego — APPROVED** [review]

**Pull request merged**